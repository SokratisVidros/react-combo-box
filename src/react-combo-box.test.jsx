import React from 'react';
import {shallow} from 'enzyme';
import ReactCombobox from './react-combo-box';
import renderer from 'react-test-renderer';

test('<ReactCombobox/> renders correctly the combobox UI with primary and alternate values', () => {
  const component = renderer.create(
    <ReactCombobox
      name='email'
      primary='john@example.com'
      alternates={['john.doe@example.com', 'john_doe@example.com']}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('<ReactCombobox/> expands and collapses alternate values on toggle click', () => {
  const component = renderer.create(
    <ReactCombobox
      name='email'
      primary='john@example.com'
      alternates={['john.doe@example.com', 'john_doe@example.com']}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Expand
  tree.children[1].props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Collapse
  tree.children[1].props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('<ReactCombobox/> updates primary value on "Make primary" click', () => {
  const cb = jest.fn();
  const component = shallow(
    <ReactCombobox
      name='email'
      primary='john@example.com'
      alternates={['john.doe@example.com', 'john_doe@example.com']}
      collapsed={false}
      onPrimaryUpdated={cb}
    />
  );

  component.find('[data-ui="make-primary-btn"]').first().simulate('click', {
    preventDefault() {},
    target: {
      getAttribute() {
        return 'john.doe@example.com';
      }
    }
  });

  expect(cb.mock.calls[0][0]).toBe('john.doe@example.com');
});

test('<ReactCombobox/> updates primary value on input change', () => {
  const cb = jest.fn();
  const component = shallow(
    <ReactCombobox
      name='email'
      primary='john@example.com'
      alternates={['john.doe@example.com', 'john_doe@example.com']}
      onPrimaryUpdated={cb}
    />
  );

  component.find('input').first().simulate('change', {
    preventDefault() {},
    target: {
      value: 'carla.coe@example.com'
    }
  });

  expect(cb.mock.calls[0][0]).toBe('carla.coe@example.com');
});

test('<ReactCombobox/> renders correctly the combobox UI without alternate values', () => {
  const component = renderer.create(
    <ReactCombobox
      name='email'
      primary='john@example.com'
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
