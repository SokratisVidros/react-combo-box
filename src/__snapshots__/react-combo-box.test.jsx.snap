// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReactCombobox/> expands and collapses alternate values on toggle click 1`] = `
<div
  className={undefined}
>
  <input
    className=""
    name="email"
    onChange={[Function]}
    value="john@example.com"
  />
  <span
    onClick={[Function]}
  >
    <span
      className=""
      data-ui="show-more"
    >
      Show 2 more
    </span>
  </span>
</div>
`;

exports[`<ReactCombobox/> expands and collapses alternate values on toggle click 2`] = `
<div
  className={undefined}
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <input
      className=""
      name="email"
      onChange={[Function]}
      value="john@example.com"
    />
    <div
      className=""
      data-ui="alternates"
    >
      <span
        className=""
        data-ui="alternate-value"
      >
        john.doe@example.com
      </span>
      <span
        className=""
        data-ui="make-primary-btn"
        data-value="john.doe@example.com"
        onClick={[Function]}
      >
        Make primary
      </span>
    </div>
    <div
      className=""
      data-ui="alternates"
    >
      <span
        className=""
        data-ui="alternate-value"
      >
        john_doe@example.com
      </span>
      <span
        className=""
        data-ui="make-primary-btn"
        data-value="john_doe@example.com"
        onClick={[Function]}
      >
        Make primary
      </span>
    </div>
  </div>
  <span
    onClick={[Function]}
  >
    <span
      className=""
      data-ui="show-less"
    >
      Show less
    </span>
  </span>
</div>
`;

exports[`<ReactCombobox/> expands and collapses alternate values on toggle click 3`] = `
<div
  className={undefined}
>
  <input
    className=""
    name="email"
    onChange={[Function]}
    value="john@example.com"
  />
  <span
    onClick={[Function]}
  >
    <span
      className=""
      data-ui="show-more"
    >
      Show 2 more
    </span>
  </span>
</div>
`;

exports[`<ReactCombobox/> renders correctly the combobox UI with primary and alternate values 1`] = `
<div
  className={undefined}
>
  <input
    className=""
    name="email"
    onChange={[Function]}
    value="john@example.com"
  />
  <span
    onClick={[Function]}
  >
    <span
      className=""
      data-ui="show-more"
    >
      Show 2 more
    </span>
  </span>
</div>
`;

exports[`<ReactCombobox/> renders correctly the combobox UI without alternate values 1`] = `
<div
  className={undefined}
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <input
      className=""
      name="email"
      onChange={[Function]}
      value="john@example.com"
    />
  </div>
</div>
`;
